sycl/test-e2e/Graph/Explicit/add_node_while_recording.cpp

```diff
@@ -1,8 +1,8 @@
 // REQUIRES: cuda || level_zero, gpu
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
-// Extra run to check for leaks in Level Zero using ZE_DEBUG
-// RUN: %if ext_oneapi_level_zero %{env ZE_DEBUG=4 %{run} %t.out 2>&1 | FileCheck %s %}
+// Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
+// RUN: %if ext_oneapi_level_zero %{env UR_L0_LEAKS_DEBUG=1 %{run} %t.out 2>&1 | FileCheck %s %}
 //
 // CHECK-NOT: LEAK
```

Comment:
Why are we doing this explicitly in some (many) tests? I thought we run **all** tests twice, and one of the runs is with ZE_DEBUG=4, this way detecting leak in all tests.

Reply:
@smaslov-intel : I guess that is a question for the team running or designing the tests? Here I have changed only the references to ZE_DEBUG=4 to UR_L0_LEAKS_DEBUG=1, but whether they should be being used at all is something they are better positioned to answer. Let me know what you think.

Reply:
Fair point, I will approve this change. But I'd like to hear from @intel/sycl-graphs-reviewers I guess about the motivation here.

Reply:
Another point is that we should synchronize with how we run this testing for leaks, it should now use UR_L0_DEBUG instead of ZE_DEBUG.

Reply:
Agree, this will help us start moving to UR env vars.

Reply:
> I thought we run all tests twice, and one of the runs is with ZE_DEBUG=4, this way detecting leak in all tests.

We introduced this leak checking when we previously had a leak of L0 events in our UR command-buffer backend for graphs, and wanted to prevent this from regressing (or issues being introduced in new tests). Crucially, this was also at a time when we were developing on a fork of DPC++, so didn't have CI to catch non-default test configurations and wanted to catch the leaks during local development.

Could you point me to where we are running the tests a second time with `ZE_DEBUG=4` (or setting `ur_l0_leaks_debug` to pass to lit)? I think the suspicion that the graphs tests don't need their own extra leak checks is probably correct if that's the case.

Reply:
thanks @EwanC .

@smaslov-intel : i think that question would be for you:

>Could you point me to where we are running the tests a second time with ZE_DEBUG=4 (or setting ur_l0_leaks_debug to pass to lit)? I think the suspicion that the graphs tests don't need their own extra leak checks is probably correct if that's the case.

Reply:
To help unblock this discussion, if you know where this second run is happening @smaslov-intel,  then create a GitHub Issue with that information and assign it to me, and I'll make this change to the graphs to remove the leak checking run from the graphs E2E tests.